src/screens/PostScreen.tsx

```diff
@@ -88,31 +88,39 @@ const styles = StyleSheet.create({
 export default function PostScreen() {
 	const [selectedImage, setSelectedImage] = useState('');
 	const [selectedImageId, setSelectedImageId] = useState('');
-	const [loading, setLoading] = useState(false);
 	const [postDescription, setPostDescription] = useState('');
-	const [pointsData, setPointsData] = useState<PointToClaim[]>([]);
+	const { isLoading, data: pointsData } = usePointToClaimQuery();
 	const [modalVisible, setModalVisible] = useState(false);
-	const [selectedPoint, setSelectedPoint] = useState(null);
+	const [selectedPoint, setSelectedPoint] = useState<number | null>(null);
 	const selectedPointData = pointsData?.find((item) => item.id === selectedPoint);
 	const { session } = useAppContext();
+	const createPostMutation = useCreateUserPostsMutation();
+	const createPointMutation = useCreateUserPointsMutation();
 
 	const createPost = async () => {
 		const imageUrl = await getImageUrl(selectedImageId);
 
 		// Create post
-		const { error } = await supabase.from('post').insert({
-			id: 55, // @TODO fix supabase not generating serial ids or use uuid
-			title: 'Post test',
-			description: postDescription,
-			user_id: session?.user.id,
-			image_url: imageUrl,
+		const post = await createPostMutation.mutateAsync({
+			id: session?.user.id as string,
+			payload: {
+				title: 'Post test',
+				description: postDescription,
+				user_id: session?.user.id as string,
+				image_url: imageUrl,
+			}
 		});
 
-		if (error) {
-			throw error;
+		if (selectedPoint) {
+			await createPointMutation.mutateAsync({
+				payload: {
+					post_id: post.id,
+					claimed_at: new Date().toISOString(),
```

Comment:
Think this might alsio be automatically generated in Supabase, but not sure

Reply:
<img width="635" alt="image" src="https://user-images.githubusercontent.com/55079408/209725521-37976aff-866a-40c0-9b9e-005d39a81710.png">
Yeah, you are right!